### ISSUE TITLE
NPE during compilation of a nulling-out `lazy val` in a non-native JS class

### ISSUE THREAD

**Issue by benhutchison** [scala-js/scala-js#3422]
```
[error] java.lang.NullPointerException
[error] 	at scala.reflect.internal.Definitions$DefinitionsClass.isPrimitiveValueType(Definitions.scala:1378)
[error] 	at org.scalajs.core.compiler.GenJSCode$JSCodePhase.ensureBoxed(GenJSCode.scala:3979)
[error] 	at org.scalajs.core.compiler.GenJSCode$JSCodePhase.genBoxedRhs$1(GenJSCode.scala:2012)
[error] 	at org.scalajs.core.compiler.GenJSCode$JSCodePhase.genStatOrExpr(GenJSCode.scala:2021)
...
```

To reproduce:
1. checkout https://github.com/benhutchison/metadoc
2. run `js/compile`

**Comment by sjrd** [comment]
Thanks for the report. Any chance you could try and minimize the bug?

**Comment by benhutchison** [comment]
Ive been able to narrow it right down to a [`lazy` modifier here](https://github.com/benhutchison/metadoc/blob/npe_minimize/metadoc-js/src/main/scala/metadoc/MetadocEditorService.scala#L21) :)

With that removed, the NPE stops.

**Comment by sjrd** [comment]
OK, well. I'll have a look next week, when I'm back from vacation. Or the week after if next week ends up too crowded.

**Comment by sjrd** [comment]
Minimization:
```scala
class Bug3422(initStr: String) extends js.Object {
  lazy val str: String = initStr
}
```
i.e., this happens
* in a non-native JS class
* with a `lazy val` whose right-hand-side refers to a class param accessor that is not otherwise used

For such `lazy val`s, scalac generates some code that nulls out `initStr` in the `str$lzycompute()` method (after it's been used to compute `str`), for garbage-collection purposes. This generated code triggers the bug. The nulling-out statement
```scala
Bug3422.this.initStr = null;
```
has an rhs which is a literal `null`, whose `tpe` is `null`. This is clearly scalac's fault, but we'll have to work around it, as usual.

**Comment by sjrd** [comment]
Fixed in 9d6078af1f6eeadc75101fd8490f32472b79c130.
